### PR TITLE
Improve Dockerfile

### PR DESCRIPTION
**Pull request by C0ffeeCode**
### Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Copy package.json and package-lock.json first and get dependencies before getting the rest
  - Avoids reinstalling dependencies when they have not been changed but only code
  - Faster build times
- Use npm ci over npm install
  - Faster build times
  - Creates more expectable results
- Stop adding "wasm" to nginx mime.types
  - nginx:latest includes it by default (now) so it just gets duplicated which results in a warning
  - This avoids that warning
- Stop copying .git/ and node_modules/ to docker image
  - Avoid unintended rebuilds

#### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
  - Warning
- [x] Improve quality (non-breaking change which fixes an issue)

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,18 +3,16 @@ FROM node:14-alpine as builder
...
+RUN npm run build
 
 
 ## App
 FROM nginx:alpine
 
 COPY --from=builder /src/dist /app
 
-# Insert wasm type into Nginx mime.types file so they load correctly.
-RUN sed -i '3i\ \ \ \ application/wasm wasm\;' /etc/nginx/mime.types
```
Comment (GoliathLabs) [inline Dockerfile:17]: Why did you remove this line though?

Comment (GoliathLabs) [inline Dockerfile:17]: Nevermind, reading the PR description explains everything :P

**Comment by GoliathLabs** [comment]
@C0ffeeCode Can you change the node version to `14.18.3-alpine` and `nginx:alpine` to `nginx:1.21.6-alpine` so Dependabot can take care of finer grained updates in the future #260 ?

**Comment by C0ffeeCode** [comment]
Sure, although I now chose a more recent version of node (17.4.0-alpine3.15).
Should I change it?

**Review by kfiven — APPROVED** [review]
LGTM
